sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -125,7 +125,6 @@ object DateTimeUtils {
       .getOrElseUpdate(timeZone, {
         Calendar.getInstance(timeZone)
       })
-    c.clear()
```

Comment:
@viirya @kiszk Do you agree with this commit?

Reply:
Seems `setTimeInMillis` can result in all fields set. If so, `clear` is redundant.